File: Python/08_MiniMaxSum.py
# -*- coding: utf-8 -*-
'''
Created on Tue Mar 01 2022

@author: Carlos Páez
'''

# 
# Complete the `miniMaxSum` function below
# 
# The function accepts INTEGER_ARRAY ass as parameter
# 

def miniMaxSum(arr):
    """
    Given an array of integers, find the minimum and maximum values that can be calculated by summing
    exactly four of the five integers. 
    Then print the respective minimum and maximum values as a single line of two space-separated long
    integers
    
    :param arr: an array of 5 integers
    """
    arr_sort = sorted(arr)
    mini, max = 0, 0
    i, j = 0, len(arr) - 1 

    if len(arr) > 5: pass

    for _ in range(len(arr) - 1):
        mini += arr_sort[i]
        max += arr_sort[j]
        i += 1
        j -= 1
    
    print(mini, max)


if __name__ == '__main__':
    arr = list(map(int, input().rstrip().split()))

    miniMaxSum(arr)

File: TypeScript/src/08-Mini_Max_Sum.ts
/**
 * Created on Mon Apr 18 2022
 * 
 * @author Carlos Páez
 */
'use strict'

process.stdin.setEncoding('utf-8')


let inputString: string = ''
let currentLine: number = 0

/* Reading the input from the user and storing it in the variable `inputString` */
process.stdin.on('data', (inputStdin: string) => {
    inputString += inputStdin
})


/* Reading the input from the user and storing it in the variable `inputString` */
process.stdin.on('end', () => {
    inputLines = inputString.split('\n')
    inputString = ''
    main()
})


/**
 * It reads a line from the input file and returns it
 */
const readLine = () => inputLines[currentLine++]


/*
 * Complete the 'miniMaxSum' function below.
 *
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function miniMaxSum(arr: number[]): void {
    arr.sort()

    let min: number = 0
    let max: number = 0

    let x = 0
    let y = arr.length - 1

    for (let i = 0; i < arr.length - 1; i++) {
        if (1 <= arr[i] && arr[i] <= Math.pow(10, 9)) {
            min += arr[x]
            max += arr[y]
            x++
            y--
        }
    }

    console.log(`${min} ${max}`)
}

function main() {

    const arr: number[] = readLine().replace(/\s+$/g, '').split(' ').map(arrTemp => parseInt(arrTemp, 10));

    miniMaxSum(arr);
}